plugins/modules/lambda_event.py

```diff
@@ -252,13 +267,31 @@ def lambda_event_stream(module, client):
     else:
         module.fail_json(msg="Source parameter 'source_arn' is required for stream event notification.")
 
-    # check if optional sub-parameters are valid, if present
-    batch_size = source_params.get("batch_size")
-    if batch_size:
-        try:
-            source_params["batch_size"] = int(batch_size)
-        except ValueError:
-            module.fail_json(msg=f"Source parameter 'batch_size' must be an integer, found: {batch_size}")
+    if state == "present":
+        if module.params["event_source"].lower() == "sqs":
+            # Default 10. For standard queues the max is 10,000. For FIFO queues the max is 10.
+            source_params.setdefault("batch_size", 10)
```

Comment:
Do we need to set these default values or can we just not send them and let AWS set the defaults?

Reply:
We need to handle them.